iree/compiler/Dialect/Shape/Plugins/XLA/XlaHloShapeBuilder.cpp

```diff
@@ -308,6 +308,128 @@ Value rewriteSelectOp(RankedShapeType resultShape, SelectOp selectOp,
   return builder.create<MakeRankedShapeOp>(loc, resultShape, dynamicDims);
 }
 
+Value rewriteTorchIndexSelect(RankedShapeType resultShape,
+                              TorchIndexSelectOp torchIndexSelectOp,
+                              OpBuilder &builder) {
+  if (!torchIndexSelectOp) return nullptr;
+  auto loc = torchIndexSelectOp.getLoc();
+
+  int64_t resultShapeRank = resultShape.getRank();
+  auto paramsType =
+      torchIndexSelectOp.input().getType().cast<RankedTensorType>();
+  auto indicesType =
+      torchIndexSelectOp.index().getType().cast<RankedTensorType>();
+  if (!paramsType || !indicesType) {
+    return nullptr;
+  }
+
+  auto axis = torchIndexSelectOp.dim();
+  auto batchDim = torchIndexSelectOp.batch_dims();
+  int64_t paramsRank = paramsType.getRank();
+  int64_t indicesRank = indicesType.getRank();
+
+  std::vector<int64_t> shape(paramsType.getShape());
+  int64_t axisValue = *(axis.getRawData());
```

Comment:
I think you want axis.getSExtValue() here and on the next line.

Reply:
done